uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/FloorFilterController.qml

```diff
@@ -23,6 +23,82 @@ QtObject {
 
     property GeoView geoView
 
+    property Connections geoViewConnections: Connections {
+        target: geoView
+        function onViewpointChanged() {
+            updateSelection()
+        }
+        // manually set when floorManager is loaded
+        enabled: false
+    }
+
+    function updateSelection() {
+        // todo: checks on floormanager not loaded/ not defined
+        // todo: swap ints with always, etc..
+        // todo: facilities and sites are optional, add checks
+        var viewpointCenter = geoView.currentViewpointCenter
+        var targetGeometry = geoView.currentViewpointExtent.extent
+        console.log("uodateselection")
+        if (floorManager.loadStatus === Enums.LoadStatusLoaded)
+            console.log(viewpointCenter.center.geometryType,
+                        floorManager.sites[0].geometry.geometryType)
+        //viewPointCenter.targetScale === double.NaN
+        if (automaticSelectionMode === FloorFilterController.AutomaticSelectionMode.Never) {
+            return
+        }
+        var targetScale = controller.floorManager.siteLayer ? controller.floorManager.siteLayer.minScale : 0
+
+        if (targetScale === 0)
+            targetScale = 4300
+        if (viewpointCenter.targetScale > targetScale) {
+            if (automaticSelectionMode === FloorFilterController.AutomaticSelectionMode.Always) {
+
+                // todo: set site, facility and level to null
+                setSelectedSiteId("")
+                setSelectedFacilityId("")
+                setSelectedLevelId("")
+            }
+            return
+        }
+
+        let selectSite
+        for (var i = 0; i < floorManager.sites.length; ++i) {
+            if (GeometryEngine.intersects(floorManager.sites[i].geometry,
+                                          viewpointCenter.center)) {
+                selectSite = floorManager.sites[i]
+                break
+            }
+        }
```

Comment:
Consider using higher-order functions when `Array` is available.
```js
const selectedSite = floorManager.sites.find(s => GeometryEngine.intersects(s.geometry, viewpointCenter.center))
```

Reply:
Sorry, that's entirely my fault. This isn't a JS `Array`,  it's a [QQmlListPoperty](https://doc.qt.io/qt-5/qqmllistproperty.html), so it doesn't have the same effective abilities. 😭

Reply:
Would https://bugreports.qt.io/browse/QTBUG-49613 solve this? This is an unfortunate pitfall of the API.

Reply:
@JamesMBallard does it mean that the functionality will come some time in the Qt 6 release?
At the moment I am transforming to an Array, but I have noticed that the JS implementation is a deep copy, so I think quite expensive to do at every viewpoint changed. I will roll it back to the traditional for loop

Reply:
@mrwillyees unlikely since both the issues are still open and categorized as merely suggestions.